pkg/apis/scheduling/validation/validation_test.go

```diff
@@ -180,3 +181,427 @@ func TestValidatePriorityClassUpdate(t *testing.T) {
 		}
 	}
 }
+
+func TestValidateWorkload(t *testing.T) {
+	successCases := map[string]*scheduling.Workload{
```

Comment:
Consider the "tweak" pattern for tests like these (which tend to grow a lot of cases).  

Write a "mkWorkload(tweaks ...Tweak)" function, then define small tweaks which allow you to mutate a valid base object into some invalid state.  It makes the tests much easier to read and manage.  For example:  https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/core/replicationcontroller/declarative_validation_test.go#L38-L189